### PR TITLE
Update build for ECE 2.7.0 release.

### PR DESCRIPTION
**Pull request by rhass**
This PR adds the conf.yaml doc build changes required for ECE 2.7.

### CODE REVIEW

File: conf.yaml
Comment (gtback) [inline conf.yaml:721]: Is there going to be a `2.6` branch that we can use to continue to make changes to the docs for that version? If not, we might need to leave it like:

```suggestion
            branches:   [ {'ms-47': 2.7}, {'release-ms-41': 2.6}, 2.5, 2.4, 2.3, 2.2, 2.1, 2.0, 1.1, 1.0 ]
```

The key is the branch name, and the value is the version name as it shows up in the URL and pages.

File: conf.yaml
Comment (gtback) [inline conf.yaml:738]:
```suggestion
                  release-ms-41: master
```
Same as above, we need to use the branch name here.

File: conf.yaml
Comment (gtback) [inline conf.yaml:753]:
```suggestion
                  release-ms-41: master
```
Same as above, we need to use the branch name here.

**Review by gtback — COMMENTED** [review]
Just double checking the branch names. If the `release-ms-41` branch isn't going away, we should probably leave it as-is.

File: conf.yaml
Comment (rhass) [reply conf.yaml:721]: Ahh, good point. I hadn't considered this when I made this change, thank you!

**Comment by rhass** [comment]
This is failing in the build, but it's not entirely clear to me why from the output.

```
21:10:53 INFO:build_docs:Bad cross-document links:
21:10:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-enterprise/1.1/ece-configuring-minio.html:
21:10:53 INFO:build_docs:   - en/cloud-enterprise/current/ece-installing-offline.html
21:10:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-enterprise/2.0/ece-configuring-minio.html:
21:10:53 INFO:build_docs:   - en/cloud-enterprise/current/ece-installing-offline.html
21:10:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-enterprise/2.1/ece-configuring-minio.html:
21:10:53 INFO:build_docs:   - en/cloud-enterprise/current/ece-installing-offline.html
21:10:53 INFO:build_docs:  /tmp/docsbuild/target_repo/html/en/cloud-enterprise/2.2/ece-configuring-minio.html:
21:10:53 INFO:build_docs:   - en/cloud-enterprise/current/ece-installing-offline.html
21:11:04 runbld>>> <<<<<<<<<<<< SCRIPT EXECUTION END <<<<<<<<<<<<
```

**Comment by gtback** [comment]
@rhass These errors aren't the most intuitive, I agree. The problem is that there are links on these pages:
- https://www.elastic.co/guide/en/cloud-enterprise/1.1/ece-configuring-minio.html
- https://www.elastic.co/guide/en/cloud-enterprise/2.0/ece-configuring-minio.html
- https://www.elastic.co/guide/en/cloud-enterprise/2.1/ece-configuring-minio.html
- https://www.elastic.co/guide/en/cloud-enterprise/2.2/ece-configuring-minio.html
That try to go to https://www.elastic.co/guide/en/cloud-enterprise/current/ece-installing-offline.html

These links work right now now, but no longer work when `cloud-enterprise/current` refers to the ms-47 branch rather than the release-ms-41 branch (which is what this PR does, in part). Presumably because that page was moved/renamed in since the last release branch. Typically @nrichers or another @elastic/cloud-writers know what has changed and how best to fix things like this.

**Review by gtback — APPROVED** [review]

**Comment by nrichers** [comment]
> These links work right now now, but no longer work when `cloud-enterprise/current` refers to the `ms-47` branch rather than the `release-ms-41` branch (which is what this PR does, in part). Presumably because that page was moved/renamed in since the last release branch. Typically @nrichers or another @elastic/cloud-writers know what has changed and how best to fix things like this.

@rhass I will create some PRs to fix these. I agree with @gtback, this is not something you should have to deal with.

**Comment by kilfoyle** [comment]
I've created PRs in the ECE 1.0, 2.0, 2.1 and 2.2 releases to fix those broken links. Once the links are fixed, this PR should be good to go.

However, I'm not sure if the CI checks for my link-fix PRs are running. They seem sluggish. @rhass would you happen to know?
https://github.com/elastic/cloud/pull/69575
https://github.com/elastic/cloud/pull/69576
https://github.com/elastic/cloud/pull/69577
https://github.com/elastic/cloud/pull/69579

**Comment by rhass** [comment]
> I've created PRs in the ECE 1.0, 2.0, 2.1 and 2.2 releases to fix those broken links. Once the links are fixed, this PR should be good to go.
> 
> However, I'm not sure if the CI checks for my link-fix PRs are running. They seem sluggish. @rhass would you happen to know?
> [elastic/cloud#69575](https://github.com/elastic/cloud/pull/69575)
> [elastic/cloud#69576](https://github.com/elastic/cloud/pull/69576)
> [elastic/cloud#69577](https://github.com/elastic/cloud/pull/69577)
> [elastic/cloud#69579](https://github.com/elastic/cloud/pull/69579)

I had the bot do a merge commit with their target branches to see if it will trigger new builds to get things flowing again. I am not really sure why these all got stuck. It's a bit odd.

**Comment by rhass** [comment]
If this remains stuck in a hour or so then let's just merge them anyway.

**Comment by kilfoyle** [comment]
I don't have the privileges to merge PRs without the CI checks having passed, so if you don't mind doing the honours, Ryan, feel free to merge them. They're just minor link fixes so I'm sure they're fine. Thanks a lot for your help!

**Comment by rhass** [comment]
@kilfoyle It seems I do not either. 😞 I'll try closing and re-opening one just to see if that helps.

**Comment by rhass** [comment]
@kilfoyle We may have to open this against master and backport.

**Comment by kilfoyle** [comment]
@rhass, I'm happy to try that, but the problem link is coded correctly in master, so I think the auto-backports wouldn't work due to a merge conflict. I could try making a minor change to the file in master and then creating backport PRs manually, but wouldn't I just run into the same problem?

@nrichers, I know you're sometimes able to merge PRs that others can't. Would you mind checking if you can merge these ones?
elastic/cloud#69575
elastic/cloud#69576
elastic/cloud#69577
elastic/cloud#69579

**Review by nrichers — APPROVED** [review]
LGTM! These changes will go live within 30 - 60 mins of merging, so please make sure the release is on track before hitting the merge button.

**Comment by kilfoyle** [comment]
@elasticmachine update branch

**Comment by kilfoyle** [comment]
Thanks to @nrichers for merging those link fix PRs! I'm running the "update branch" to try and relaunch the checks on this PR.

**Comment by kilfoyle** [comment]
jenkins test this please

**Pull request merged**
